ethereum/contracts/ERC20App.sol

```diff
@@ -73,20 +79,39 @@ contract ERC20App is AccessControl {
             "Invalid channel ID"
         );
 
+        uint8 _decimals;
+        string memory _name;
+        string memory _symbol;
+
+        (_name, _symbol, _decimals) = tokenDetails(_token);
```

Comment:
this looks like it could be within the if statement below, so that it only runs the first time, right?

Reply:
Yes,exactly it should be within the if condition only.